### PR TITLE
Update lava-optimization dependencies

### PR DESCRIPTION
**Pull request by mgkwill**

### CODE REVIEW

**Review by GaboFGuerra — APPROVED** [review]
Thanks Marcus, looks good!

File: poetry.lock [hunk 1]
```diff
@@ -276,6 +296,43 @@ category = "dev"
...
+category = "main"
+optional = false
+python-versions = ">=3.6"
+
+[package.dependencies]
+dimod = ">=0.10.0,<0.11.0"
+numpy = ">=1.17.3,<2.0.0"
+
+[[package]]
+name = "dwave-qbsolv"
```
Comment (phstratmann) [inline poetry.lock:326]: May be left out - see above.

File: poetry.lock [hunk 2]
```diff
@@ -276,6 +296,43 @@ category = "dev"
...
+[[package]]
+name = "dwave-preprocessing"
+version = "0.3.2"
+description = "An API for preprocessing tools"
+category = "main"
+optional = false
+python-versions = ">=3.6"
+
+[package.dependencies]
+dimod = ">=0.10.0,<0.11.0"
```
Comment (phstratmann) [inline poetry.lock:322]: In this release, we won't yet include any pipeline to benchmark against Pyomo / QBSolv yet. Thus, we may leave out the requirement to install...
... dimod
... dwave-qbsolv
...pyomo

File: poetry.lock [hunk 3]
```diff
@@ -276,6 +296,43 @@ category = "dev"
...
+[[package]]
+name = "dwave-qbsolv"
+version = "0.3.4"
+description = ""
+category = "main"
+optional = false
+python-versions = ">=3.6"
+
+[package.dependencies]
+dimod = ">=0.8.1,<0.11.0"
```
Comment (phstratmann) [inline poetry.lock:334]: dimod may be left out - see above.

File: poetry.lock [hunk 4]
```diff
@@ -1190,6 +1274,22 @@ category = "dev"
 optional = false
 python-versions = "*"
 
+[[package]]
+name = "pyomo"
```
Comment (phstratmann) [inline poetry.lock:1278]: May be left out - see above.

File: poetry.lock [hunk 5]
```diff
@@ -1722,7 +1883,82 @@ defusedxml = [
     {file = "defusedxml-0.7.1-py2.py3-none-any.whl", hash = "sha256:a352e7e428770286cc899e2542b6cdaedb2b4953ff269a210103ec58f6198a61"},
     {file = "defusedxml-0.7.1.tar.gz", hash = "sha256:1bb3032db185915b62d7c6209c5a8792be6a32ab2fedacc84e01b52c51aa3e69"},
 ]
+dimod = [
```
Comment (phstratmann) [inline poetry.lock:1886]: May be left out - see above.

File: poetry.lock [hunk 6]
```diff
@@ -1722,7 +1883,82 @@ defusedxml = [
...
+    {file = "dwave_preprocessing-0.3.2-cp38-cp38-manylinux_2_17_aarch64.manylinux2014_aarch64.whl", hash = "sha256:38c05e5a29fe1cc4c386045481a4aa07564bd0447c42e276e71459c72eff5753"},
+    {file = "dwave_preprocessing-0.3.2-cp38-cp38-win_amd64.whl", hash = "sha256:f3f29f754a91a291bb0f793cf6066d23638e53335a8d4b1dd77f6262e13129a2"},
+    {file = "dwave_preprocessing-0.3.2-cp39-cp39-macosx_10_9_universal2.whl", hash = "sha256:a523351a95bf4e22c0724177ba587f3107755cdd4556046962cf5cdc1515272b"},
+    {file = "dwave_preprocessing-0.3.2-cp39-cp39-macosx_10_9_x86_64.whl", hash = "sha256:587499d0bcabf30bd7310ec3fb9f3177f84982c1843442cdf521452ab9d67209"},
+    {file = "dwave_preprocessing-0.3.2-cp39-cp39-manylinux_2_12_x86_64.manylinux2010_x86_64.whl", hash = "sha256:9f7256607a54465bcd844d83aae5f19a4730eed56dc4eda8ccffccbcf0efe42a"},
+    {file = "dwave_preprocessing-0.3.2-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl", hash = "sha256:59a00241094ca744cb74c4d1a411928890118da9618993d76128c65859dd0091"},
+    {file = "dwave_preprocessing-0.3.2-cp39-cp39-win_amd64.whl", hash = "sha256:0fe8bad4f6b05980a6f0c58e45020a05da14569bf6ff9f860825464de1198cee"},
+]
+dwave-qbsolv = [
```
Comment (phstratmann) [inline poetry.lock:1940]: May be left out - see above.

File: poetry.lock [hunk 7]
```diff
@@ -2271,6 +2519,21 @@ pylev = [
     {file = "pylev-1.4.0-py2.py3-none-any.whl", hash = "sha256:7b2e2aa7b00e05bb3f7650eb506fc89f474f70493271a35c242d9a92188ad3dd"},
     {file = "pylev-1.4.0.tar.gz", hash = "sha256:9e77e941042ad3a4cc305dcdf2b2dec1aec2fbe3dd9015d2698ad02b173006d1"},
 ]
+pyomo = [
```
Comment (phstratmann) [inline poetry.lock:2522]: May be left out - see above.

File: pyproject.toml [hunk 8]
```diff
@@ -51,6 +51,13 @@ lava-nc = { git = "https://github.com/lava-nc/lava.git", branch = "main", develo
 numpy = "^1.23.1"
 pytest = "^7.1.2"
 unittest2 = "^1.1.0"
+Pyomo = "^6.4.2"
+networkx = "<=2.8"
+dsolve = "^0.0.5"
+dwave-qbsolv = "^0.3.4"
```
Comment (phstratmann) [inline pyproject.toml:57]: May be left out - see above.

**Review by phstratmann — CHANGES_REQUESTED** [review]
Due to time constraints, the benchmarking utility will not make it into the release. Thus, we may leave out the dependencies on...
dimod
dwave-qbsolv
pyomo.

Apart from that, looks good, thx!

**Pull request merged**